### PR TITLE
Use icon-more in breadcrumb for unshared folders instead of icon-share, fix #3564

### LINKED ISSUE #3564: New share icon in breadcrumb nav
The new share icon in the breadcrumb navigation was probably meant to speed up everyones workflow. As a user I have to say that I find it **very** confusing.

bildschirmfoto 2017-02-21 um 16 12 17
*not shared folder*

bildschirmfoto 2017-02-21 um 16 12 35
*shared folder*

My points of criticism being:
- The **difference** between the shared/not shared state is **not big enough**. The not-shared state looks like the shared state if no comparison is available. I panicked seeing this the first time, thinking the folder was shared.
- Why is the icon **only shown for the current folder**? This fact could support the confusion, because not every folder in the hierarchy has the icon, which could suggest that this particular folder is shared – while it's not (this scenario is not visible in the screenshots).
- When I click on the icon I get to the details panel of the folder I'm currently in. This icon is the only option to get there **from within** the folder. But there are other tabs in the panel (comment, activities). How should a user reach them from inside the folder? It seems inconsistent to provide an icon for one action, but not for all of them.

My suggestion:
Well, I don't have one at the moment. Personally, I see this as a big cause of confusion within the interface, because of the three inconsistencies mentioned. I'm interested, if anyone agrees to that – or if maybe someone might be able to come up with a solution (that is not removing the icon again).

### PR DESCRIPTION
**Pull request by jancborchardt**
Please review @MorrisJobke @te-online 

Non-shared folders are now not using the share icon anymore but rather the 3-dot icon to open the sidebar. This prevents the confusion that you could think the folder is already shared.
screenshot from 2017-08-16 00-53-13
Actually shared folder, as usual:
screenshot from 2017-08-16 00-53-27

The only possible issue I see is that there’s a 3-dot icon right above, in the navigation. However it’s kinda the same use »show more« and it only appears if you have a lot of apps installed so maybe a non-issue. cc @nextcloud/designers 

fix #3564

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Two tests fail:

```
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.ShareBreadCrumbView tests Rendering Render in dir FAILED
	Expected 0 to equal 1.
	apps/files_sharing/tests/js/sharedbreadcrumviewSpec.js:74:54
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.ShareBreadCrumbView tests Rendering Remove shared status from user share FAILED
	Expected 0 to equal 1.
	apps/files_sharing/tests/js/sharedbreadcrumviewSpec.js:202:54
```

**Comment by MorrisJobke** [comment]
I updated the tests and also added some checks to the other tests.

**Review by MorrisJobke — APPROVED** [review]
Tested and works nicely 👍

**Comment by MorrisJobke** [comment]
cc @nextcloud/designers please review

**Comment by jancborchardt** [comment]
@MorrisJobke thanks! What do you think about the change?

**Comment by MorrisJobke** [comment]
> What do you think about the change?

As stated: I like it and it looks nice.

**Comment by skjnldsv** [comment]
I really like it, but could we add a tooltip on this? :)

**Review by skjnldsv — APPROVED** [review]

**Comment by pixelipo** [comment]
I was always wondering why "share" was so prioritized folder action that it required own button in breadcrumbs, so this is a step up.

I am not, however, sold on the icon being used. 3dots usually brings up a popover menu, not sidebar. This icon is associated with sidebar:
image

File: apps/files_sharing/css/sharebreadcrumb.scss [hunk 1]
```diff
@@ -21,7 +21,8 @@
  */
 
 div.crumb span.icon-shared,
-div.crumb span.icon-public {
+div.crumb span.icon-public,
```
Comment (pixelipo) [inline apps/files_sharing/css/sharebreadcrumb.scss]: time for `div.crumb span[class^="icon-"]` here?

File: apps/files_sharing/js/sharebreadcrumbview.js [hunk 2]
```diff
@@ -44,7 +44,7 @@
 
 			if (this._dirInfo !== null && (this._dirInfo.path !== '/' || this._dirInfo.name !== '')) {
 				var isShared = data.dirInfo && data.dirInfo.shareTypes && data.dirInfo.shareTypes.length > 0;
-				this.$el.removeClass('shared icon-public icon-shared');
+				this.$el.removeClass('shared icon-public icon-shared icon-more');
```
Comment (pixelipo) [inline apps/files_sharing/js/sharebreadcrumbview.js]: `this.$el.removeClass();` should also work in this case and no worries later

File: apps/files_sharing/js/sharebreadcrumbview.js [hunk 3]
```diff
@@ -53,12 +53,12 @@
...
 					}
 				} else {
-					this.$el.addClass('icon-shared');
+					this.$el.addClass('icon-more');
 				}
 				this.$el.show();
 				this.delegateEvents();
 			} else {
-				this.$el.removeClass('shared icon-public icon-shared');
+				this.$el.removeClass('shared icon-public icon-shared icon-more');
```
Comment (pixelipo) [inline apps/files_sharing/js/sharebreadcrumbview.js]: same as above

**Comment by te-online** [comment]
I love it, thank you! 🎉 😃 

Found one bug, though, maybe it's supposed to be like this? Steps:
1. Create a new folder in a subfolder
2. The sidebar opens and that folder is selected, now click on “Share by link”
3. Now in the breadcrumbs we see the share button instead of the options button on this parent folder we are in.
4. Change to a different folder and come back to this “parent” folder: The options buttons is there again, as intended.

I don't think this is breaking, but could be worth fixing 😉

File: apps/files_sharing/css/sharebreadcrumb.scss (hunk 1 above)
Comment (jancborchardt) [inline apps/files_sharing/css/sharebreadcrumb.scss]: I’d say the three selectors are faster than this catchall, no?

File: apps/files_sharing/js/sharebreadcrumbview.js (hunk 2 above)
Comment (jancborchardt) [inline apps/files_sharing/js/sharebreadcrumbview.js]: Might not be so futureproof though if we add more classes, right?

File: apps/files_sharing/css/sharebreadcrumb.scss (hunk 1 above)
Comment (pixelipo) [inline apps/files_sharing/css/sharebreadcrumb.scss]: In general yes, but `div.crumb span.icon-public` is about as slow as single attribute selector. If you're worried about performance, you should add a new class that would apply just for that span and use it without all the cruft in front.
`.crumb [class^="icon-"]` is probably a little bit slower than the above, `.crumb .icon-public` a bit faster.

File: apps/files_sharing/js/sharebreadcrumbview.js (hunk 2 above)
Comment (pixelipo) [inline apps/files_sharing/js/sharebreadcrumbview.js]: You've got a point, if we want to add an "always on" class. Though in that case this whole thing might be worth rewriting.

In fact, I just noticed another thing - there is no more need for `.shared` selector any more. It was used to change opacity of the `icon-shared`, which had the meaning of "on/off". Now that `.icon-more` represents the "off" state, it doesn't make sense the fade it out to 0.2 opacity (also not needed).

**Comment by rullzer** [comment]
@jancborchardt what is left here?

**Comment by jancborchardt** [comment]
@pixelipo do you want to add a commit with the changes as you see fit? :)

**Comment by rullzer** [comment]
Rebased onto master

**Comment by pixelipo** [comment]
I'll have a look at it, @jancborchardt - I just want to check if we can get rid of the `.shared` selector. If not, I'll approve this PR.

**Comment by jancborchardt** [comment]
I’m actually not sure about this anymore when using it. The share icon is pretty clear in what it does. The 3-dot icon we already use in the top bar for apps overflow, and for the file actions (in both cases opening a menu). Also, there will be some »…« for when the breadcrumbs overflow.

I would stick with the share icon for now and close this.

**Comment by jancborchardt** [comment]
I’m closing this especially because the new breadcrumb overflow will also use the 3-dot icon. #7051

**Pull request closed without merging**